File: service/src/test/java/no/nav/melosys/service/kontroll/feature/ufm/kontroll/UfmKontrollsettTest.java
package no.nav.melosys.service.kontroll.feature.ufm.kontroll;

import java.util.Set;
import java.util.function.Function;

import no.nav.melosys.domain.eessi.SedType;
import no.nav.melosys.domain.kodeverk.begrunnelser.Kontroll_begrunnelser;
import no.nav.melosys.service.kontroll.feature.ufm.data.UfmKontrollData;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

class UfmKontrollsettTest {

    @Test
    void hentKontrollerA001_verifiserKontroller() {
        Set<Function<UfmKontrollData, Kontroll_begrunnelser>> kontroller =
            UfmKontrollsett.hentRegelsettForSedType(SedType.A001);
        assertThat(kontroller).hasSize(10);
    }

    @Test
    void hentKontrollerA003_verifiserKontroller() {
        Set<Function<UfmKontrollData, Kontroll_begrunnelser>> kontroller =
            UfmKontrollsett.hentRegelsettForSedType(SedType.A003);
        assertThat(kontroller).hasSize(11);
    }

    @Test
    void hentKontrollerA009_verifiserKontroller() {
        Set<Function<UfmKontrollData, Kontroll_begrunnelser>> kontroller =
            UfmKontrollsett.hentRegelsettForSedType(SedType.A009);
        assertThat(kontroller).hasSize(10);
    }

    @Test
    void hentKontrollerA010_verifiserKontroller() {
        Set<Function<UfmKontrollData, Kontroll_begrunnelser>> kontroller =
            UfmKontrollsett.hentRegelsettForSedType(SedType.A010);
        assertThat(kontroller).hasSize(10);
    }

    @Test
    void hentKontrollerA008_verifiserIngenKontroller() {
        assertThatExceptionOfType(UnsupportedOperationException.class)
            .isThrownBy(() -> UfmKontrollsett.hentRegelsettForSedType(SedType.A008))
            .withMessageContaining("A008 er ikke støttet");
    }
}

File: service/src/test/kotlin/no/nav/melosys/service/kontroll/feature/ufm/kontroll/UfmKontrollsettTest.kt
package no.nav.melosys.service.kontroll.feature.ufm.kontroll

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.matchers.collections.shouldHaveSize
import io.kotest.matchers.string.shouldContain
import no.nav.melosys.domain.eessi.SedType
import org.junit.jupiter.api.Test

class UfmKontrollsettTest {

    @Test
    fun hentKontrollerA001_verifiserKontroller() {
        val kontroller = UfmKontrollsett.hentRegelsettForSedType(SedType.A001)

        kontroller.shouldHaveSize(10)
    }

    @Test
    fun hentKontrollerA003_verifiserKontroller() {
        val kontroller = UfmKontrollsett.hentRegelsettForSedType(SedType.A003)

        kontroller.shouldHaveSize(11)
    }

    @Test
    fun hentKontrollerA009_verifiserKontroller() {
        val kontroller = UfmKontrollsett.hentRegelsettForSedType(SedType.A009)

        kontroller.shouldHaveSize(10)
    }

    @Test
    fun hentKontrollerA010_verifiserKontroller() {
        val kontroller = UfmKontrollsett.hentRegelsettForSedType(SedType.A010)

        kontroller.shouldHaveSize(10)
    }

    @Test
    fun hentKontrollerA008_verifiserIngenKontroller() {
        shouldThrow<UnsupportedOperationException> {
            UfmKontrollsett.hentRegelsettForSedType(SedType.A008)
        }.message shouldContain "A008 er ikke støttet"
    }
}
